### PR TITLE
Report analysis screen: add function highlighting

### PR DESCRIPTION
**Pull request by NoiseByNorthwest**
This patch adds the possibility, within the report analysis screen,
to highlight a function by clicking on one of its spans within the
timeline or Flamegraph widgets or on its name within the flat
profile widget.

### CODE REVIEW

File: assets/web-ui/js/widget.js [hunk 1]
```diff
@@ -453,6 +472,10 @@ class Widget {
         this.repaint();
     }
 
+    onHighlightedFunctionUpdate() {
+        this.repaint();
```
Comment (DanielRuf) [inline assets/web-ui/js/widget.js:476]: It seems on click this scrolls to the top of the list again. Just wanted to note this.

Comment (DanielRuf) [reply assets/web-ui/js/widget.js:476]: Also this prevents that I can copy the entry text via mouse (click + drag or doubleblick).

Comment (NoiseByNorthwest) [reply assets/web-ui/js/widget.js:476]: @DanielRuf yes in the Flat profile view I confirm this annoying behavior.

Comment (NoiseByNorthwest) [reply assets/web-ui/js/widget.js:476]: The "repaint" also causes the text to be unselected.

Comment (DanielRuf) [reply assets/web-ui/js/widget.js:476]: Should I file an issue for that @NoiseByNorthwest?

Comment (NoiseByNorthwest) [reply assets/web-ui/js/widget.js:476]: Yes, I'll also accept external contribution as long as the patch does not make things worst in terms of UX. It's unlikely that I'll fix this myself as I'm currently focusing on a full rewrite of the web UI which may be available within 6-12 months.

**Pull request merged**